Sources/Widgets/Representations/ResliceCursorContextRepresentation/index.js

```diff
@@ -0,0 +1,247 @@
+    updateRender(axis2State, model.pipelines.axes[1]);
+
+    publicAPI.setSphereRadius(state.getSphereRadius());
+
+    outData[0] = vtkPolyData.newInstance();
+  };
+
+  publicAPI.updateActorVisibility = (
+    renderingType,
+    wVisible,
+    ctxVisible,
+    hVisible
+  ) => {
+    if (renderingType === RenderingTypes.PICKING_BUFFER) {
+      publicAPI.getActors().forEach((actor) => actor.setVisibility(wVisible));
+    } else {
+      publicAPI
+        .getActors()
+        .forEach((actor) => actor.setVisibility(wVisible && hVisible));
+    }
+  };
+
+  publicAPI.getSelectedState = (prop, compositeID) => {
+    const state = model.inputData[0];
+    state.setActiveViewName(model.viewName);
+
+    const getAxis1 = `get${model.axis1Name}`;
+    const getAxis2 = `get${model.axis2Name}`;
+    const axis1State = state[getAxis1]();
+    const axis2State = state[getAxis2]();
+
+    let activeLineState = null;
+    let activeRotationPointName = '';
+    let methodName = '';
+
+    switch (prop) {
+      case model.pipelines.axes[0].line.actor:
+        activeLineState = axis1State;
+        methodName = 'translateAxis';
+        break;
+      case model.pipelines.axes[1].line.actor:
+        activeLineState = axis2State;
+        methodName = 'translateAxis';
+        break;
+      case model.pipelines.axes[0].rotation1.actor:
+        activeLineState = axis1State;
+        activeRotationPointName = 'RotationPoint1';
+        methodName = 'rotateLine';
+        break;
+      case model.pipelines.axes[0].rotation2.actor:
+        activeLineState = axis1State;
+        activeRotationPointName = 'RotationPoint2';
+        methodName = 'rotateLine';
+        break;
+      case model.pipelines.axes[1].rotation1.actor:
+        activeLineState = axis2State;
+        activeRotationPointName = 'RotationPoint1';
+        methodName = 'rotateLine';
+        break;
+      case model.pipelines.axes[1].rotation2.actor:
+        activeLineState = axis2State;
+        activeRotationPointName = 'RotationPoint2';
+        methodName = 'rotateLine';
+        break;
+      default:
+        methodName = 'translateCenter';
+        break;
+    }
+
+    state.setActiveLineState(activeLineState);
+    state.setActiveRotationPointName(activeRotationPointName);
+    state.setUpdateMethodName(methodName);
+
+    return state;
+  };
+}
+
+// ----------------------------------------------------------------------------
+// Object factory
+// ----------------------------------------------------------------------------
+
+const DEFAULT_VALUES = {
+  axis1Name: '',
+  axis2Name: '',
+  viewName: '',
+};
+
+// ----------------------------------------------------------------------------
+
+export function extend(publicAPI, model, initialValues = {}) {
+  Object.assign(model, DEFAULT_VALUES, initialValues);
+
+  vtkWidgetRepresentation.extend(publicAPI, model, initialValues);
+
+  model.image = null;
+  model.center = [0, 0, 0];
```

Comment:
still needed ?

Reply:
It's still a good way to initialize attribute, isn't it ?

Reply:
There are other places where it can be initialized (e.g. in DEFAULT_VALUES).

But it seems that model.center and model.image are never used, are they?

Reply:
My bad. I thought it was in the ResliceCursorWidget/index.js file. Yes, indeed we can remove it.